src/sentry/api/endpoints/project_release_files.py

```diff
@@ -125,7 +125,22 @@ def post_releasefile(request, release, logger):
                 {"detail": "File name must not contain special whitespace characters"}, status=400
             )
 
+        headers = {"Content-Type": fileobj.content_type}
+        for headerval in request.data.getlist("header") or ():
+            try:
+                k, v = headerval.split(":", 1)
+            except ValueError:
+                return Response({"detail": "header value was not formatted correctly"}, status=400)
+            else:
+                if _filename_re.search(v):
+                    return Response(
+                        {"detail": "header value must not contain special whitespace characters"},
+                        status=400,
+                    )
+                headers[k] = v.strip()
```

Comment:
I assume moving this block doesn't change any functionality?

Reply:
No it shouldn’t.
I plan to change this function so that it can also create `ArtifactBundle`s with a feature flag.